### ISSUE TITLE
Localization improvement

### ISSUE THREAD

**Issue by devmes** [sourcebroker/t3api#63]
I use the branch "typo3-115-support" in a TYPO3 v11 installation.
This works fine, but I have some problems with the "x-locale" header.

Not all content is translated correctly. For example urls are not translated and always point to the default language.
Also, in the API URL the prefix of the default language is always visible e.g. "example.com/en/_api/...". However, this is just an inconvenience.

Suggested solution:
There are two ways how to set the language correctly:
- Using "x-locale" header, then there is no need for language prefix in API url. e.g. "example.com/_api/..."
- With language prefix in API url. e.g. "example.com/en/_api/..." , "example.com/it/_api/..."

I think it would be better to set the language not in the "Language Processor" but in a separate middleware before the TSFE initialization.
This will fix the translation problem and set the language correctly in the TSFE and in the request. 
I made some changes in a [fork of the typo3-115 branch](https://github.com/devmes/t3api/commit/1098a6cdf5229cbf808ede29e3f39983516bedd4), so that by default the LanguagePrefix is removed from the API url.
The language can be set using "x-locale" header.
It also still works for backwards compatibility API Urls with LanguagePrefix.
The x-locale header only responds to the default language.

For me this solution works fine, but I am not sure if this is the best solution.

It would be helpful if this functionality could be added to the extension.

**Comment by pixelbrackets** [comment]
**tl;dr** The patch helped us well, would love to see this in T3API.

We stumbled upon this issue in two setups.

First setup uses several languages. All have its own language path (so no language uses »/« als base). T3API falls back to the default langauge for all URIs. Which is `/de` in our case. So we then fetch content from `/de/_api/*` while browsing the `/en/` pagetree. This technically works, but as described above, it confusing for developers and testers trying to keep both language paths apart. Eg. all JavaScript react to the current language, except for api calls, which are hardcoded to use the default langauge instead.

The second setup uses one domain »example.com« and several languages (»/de/en/pl/…«). *Some* languages, like the english translation, have its own domain as baseVariant (international.example.com). T3API then tries to resolve the default language of the site config again ([getDefaultLanguageBasePath](https://github.com/sourcebroker/t3api/blob/9faf7ea61bb9abf6ba5fff098e3a7debad8dbe37/Classes/Service/RouteService.php#L59)), regardless of the currently used language or domain. This causes T3API trying to match `/de/_api/*` while using the english domain. However, »de« is not available here. So all API request fail.

The patch written by @devmes resolves both situations. Every language may call `/[language path]/_api/` and is working fine.

**Comment by kszymukowicz** [comment]
Thanks for your improvments.
This is now supported in future release https://github.com/sourcebroker/t3api/tree/release/3.0.0
